### DISCUSSION TITLE
About change Task Manager UI

### DISCUSSION THREAD

**Discussion by jc3213** [jc3213/download_with_aria2#64] [Polls]
#### Why 原因

As of issues #61 #62 #63

I've tried to change the UI layout of `Task Manager`. Here's a preview on the bottom
我准备下一个版本更换`任务管理器`的布局，底部是预览

#### Download 下载
[popup_v1.zip](https://github.com/jc3213/download_with_aria2/files/11550168/popup_v1.zip)
[popup_v2.zip](https://github.com/jc3213/download_with_aria2/files/11552268/popup_v2.zip)

#### Difference 区别
- Props
    - All tasks can be filtered based on its status
    - 所有任务可以按照其进行状态进行过滤
- Cons
   - Confusing and crowded interface, will be better when `Open in New Tab`
   - 过于拥挤而充满混乱的界面，通过启用`新标签页打开`选项可得到缓解
- Todo
    - Split `❌` into 2 buttons, there's no proper emoji though
    - 将 `❌`的功能分割成两个按钮，目前没找到合适的emoji用于其中一个功能

#### Preview 预览
<details>
<summary>Original</summary>

![Old](https://github.com/jc3213/download_with_aria2/assets/8744936/66518445-d472-4f30-9fe6-365820ee3496)
</details>
<details>
<summary>Popup v1 preview</summary>

![v1_pre1](https://github.com/jc3213/download_with_aria2/assets/8744936/c23b4bd7-855d-4201-8601-9bdd11759c06)
![v1_pre2](https://github.com/jc3213/download_with_aria2/assets/8744936/87b0ac9f-e281-443c-aa55-b1a2156acdf8)
![v1_pre3](https://github.com/jc3213/download_with_aria2/assets/8744936/0e38f289-6a38-4e41-8886-f7e59d51d074)
</details>
<details>
<summary>Popup v2 preview</summary>

![v2_pre1](https://github.com/jc3213/download_with_aria2/assets/8744936/add4c760-4961-49fe-b457-e2d32dbd04db)
![v2_pre2](https://github.com/jc3213/download_with_aria2/assets/8744936/7be4a60b-db03-44c8-934f-6f1d5b857b9a)
![v2_pre3](https://github.com/jc3213/download_with_aria2/assets/8744936/ea2a111d-1152-4d7d-80d7-5af30423b00a)
</details>

**Comment by jc3213** [comment]
@daiaji @ivysrono @wordpure @hl2guide 
Please vote for your opinion
请投上一票，谢谢喵

**Reply by ivysrono** [reply]
居然还能投票

**Comment by wordpure** [comment]
从功能性的角度看，我更喜欢新界面。

界面过于拥挤的问题，可以改用图标（文字加到 title 里）来解决。

![图片](https://github.com/jc3213/download_with_aria2/assets/68013962/a81bd4b7-3b0b-4ee2-8b0a-450111c88552)

框选出来的那一串图标，是我用 Aria2NG 界面的 FontAwesome 图标魔改的。

**Reply by jc3213** [reply]
我就是找不到适合做icon的图片，所以一直拿emoji做的。这个问题就很麻烦了。

**Reply by wordpure** [reply]
可以用免费的 [Font Awesome](https://fontawesome.com/plans)

**Reply by jc3213** [reply]
没有外置的啊？只有用js。感觉头疼。最终弄得太复杂还不如直接去用Aria2Ng

**Reply by wordpure** [reply]
[Font-Awesome/svgs at master · FortAwesome/Font-Awesome · GitHub](https://github.com/FortAwesome/Font-Awesome/tree/master/svgs)

官方的 SVG 格式图标

**Reply by jc3213** [reply]
https://jc3213.github.io/jslib/manager/index.html 目前临时做好是这样。然后作为popup的时候，不显示任务计数。

其实我准备改成Queue下拉菜单里面也显示各个任务的计数。理论上并不难就是了。

并且我增加了快捷键。 Ctrl+S打开设置，Ctrl+R清理任务，Ctrl+D打开新下载，Ctrl+Q打开筛选用的列表

**Comment by jc3213** [comment]
[测试版 Pre Release](https://github.com/jc3213/download_with_aria2/releases/tag/4.4.0.2028)

@ivysrono @hl2guide

**Reply by jc3213** [reply]
超轻量在线版 https://jc3213.github.io/jslib/manager/index.html 注意，跟测试版比界面布局略有不同。但是大同小异就是了——主要还是`Purge`按钮的位置不同
